tests/test_check.py

```diff
@@ -0,0 +1,391 @@
+import pytest
+from requests_mock import mocker
+from opensafely._vendor import requests
+from opensafely import check
+import os
+import subprocess
+
+# Because we're using a vendored version of requests we need to monkeypatch the
+# requests_mock library so it references our vendored library instead
+mocker.requests = requests
+mocker._original_send = requests.Session.send
+
+
+RESTRICTED_FUNCTION_CALL = (
+    "patients.admitted_to_icu("
+    "between=['2021-01-01','2022-02-02'], "
+    "find_first_match_in_period=True, "
+    "returning='binary_flag')"
+)
+
+UNRESTRICTED_FUNCTION_CALL = (
+    "patients.with_these_medications("
+    "between=['2021-01-01','2022-02-02'], "
+    "find_first_match_in_period=True, "
+    "returning='binary_flag')"
+)
+
+
+PERMISSIONS_TEXT = """
+opensafely/dummy_icnarc:
+    allow: ['icnarc']
+opensafely/dummy_ons:
+    allow: ['ons']
+opensafely/dummy_icnarc_ons:
+    allow: ['icnarc','ons']
+"""
+
+PERMITTED_REPO = "opensafely/dummy_icnarc"
+PERMITTED_MULTIPLE_REPO = "opensafely/dummy_icnarc_ons"
+UNPERMITTED_REPO = "opensafely/dummy_ons"
+NONEXISTENT_REPO = "opensafely/dummy"
+
+PERMITTED_HTTPS_ORIGIN = f"https://github.com/{PERMITTED_REPO}"
+PERMITTED_MULTIPLE_HTTPS_ORIGIN = (
+    f"https://github.com/{PERMITTED_MULTIPLE_REPO}"
+)
+UNPERMITTED_HTTPS_ORIGIN = f"https://github.com/{UNPERMITTED_REPO}"
+NONEXISTENT_HTTPS_ORIGIN = f"https://github.com/{NONEXISTENT_REPO}"
+
+PERMITTED_SSH_ORIGIN = f"git@github.com:{PERMITTED_REPO}.git"
+PERMITTED_MULTIPLE_SSH_ORIGIN = f"git@github.com:{PERMITTED_MULTIPLE_REPO}.git"
+UNPERMITTED_SSH_ORIGIN = f"git@github.com:{UNPERMITTED_REPO}.git"
+NONEXISTENT_SSH_ORIGIN = f"git@github.com:{NONEXISTENT_REPO}.git"
+
+
+STUDY_DEF_HEADER = """from cohortextractor import (StudyDefinition, patients)
+study = StudyDefinition ("""
+
+
+def write_restricted_files(path):
+    for a in [1, 2]:
+        with open(
+            os.path.join(str(path), f"study_definition_restricted_{a}.py"), "w"
```

Comment:
tmp_path is an instance of pathlib.Path, and is the newer preferred way to interact with the file system.
So rather than converting it to a string and using os.path.*, I'd do this:

```python
    ....
    f = path / f"study_definition_restriction_{a}.py"
    f.write_text(textwrap.dedent(f"""\
        {STUDY_DEF_HEADER}
        a={RESTRICTED_FUNCTION_CALL},
        #b={RESTRICTED_FUNCTION_CALL},
        c={UNRESTRICTED_FUNCTION_CALL},
        #d={UNRESTRICTED_FUNCTION_CALL},
    """)
    ...
```